crates/hyperswitch_connectors/src/connectors/deutschebank/transformers.rs

```diff
@@ -56,81 +128,317 @@ impl TryFrom<&DeutschebankRouterData<&PaymentsAuthorizeRouterData>>
+            | DeutschebankSEPAMandateStatus::Created
+            | DeutschebankSEPAMandateStatus::PendingApproval
+            | DeutschebankSEPAMandateStatus::PendingSecondaryApproval
+            | DeutschebankSEPAMandateStatus::PendingReview
+            | DeutschebankSEPAMandateStatus::PendingSubmission
+            | DeutschebankSEPAMandateStatus::Submitted => Self::AuthenticationPending,
+            DeutschebankSEPAMandateStatus::Failed
+            | DeutschebankSEPAMandateStatus::Discarded
+            | DeutschebankSEPAMandateStatus::Expired
+            | DeutschebankSEPAMandateStatus::Replaced => Self::Failure,
+        }
+    }
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
+pub struct DeutschebankMandatePostResponse {
+    rc: String,
+    message: String,
+    mandate_id: Option<String>,
+    reference: Option<String>,
+    approval_date: Option<String>,
+    language: Option<String>,
+    approval_by: Option<DeutschebankSEPAApproval>,
+    state: Option<DeutschebankSEPAMandateStatus>,
+}
+
+impl
+    TryFrom<
+        ResponseRouterData<
+            Authorize,
+            DeutschebankMandatePostResponse,
+            PaymentsAuthorizeData,
+            PaymentsResponseData,
+        >,
+    > for RouterData<Authorize, PaymentsAuthorizeData, PaymentsResponseData>
+{
     type Error = error_stack::Report<errors::ConnectorError>;
-    fn try_from(auth_type: &ConnectorAuthType) -> Result<Self, Self::Error> {
-        match auth_type {
-            ConnectorAuthType::HeaderKey { api_key } => Ok(Self {
-                api_key: api_key.to_owned(),
+    fn try_from(
+        item: ResponseRouterData<
+            Authorize,
+            DeutschebankMandatePostResponse,
+            PaymentsAuthorizeData,
+            PaymentsResponseData,
+        >,
+    ) -> Result<Self, Self::Error> {
+        let signed_on = match item.response.approval_date {
+            Some(date) => date.chars().take(10).collect(),
+            None => time::OffsetDateTime::now_utc().date().to_string(),
+        };
+        Ok(Self {
+            status: if item.response.rc == "0" {
+                match item.response.state {
+                    Some(state) => common_enums::AttemptStatus::from(state),
+                    None => common_enums::AttemptStatus::Failure,
+                }
+            } else {
+                common_enums::AttemptStatus::Failure
+            },
+            response: Ok(PaymentsResponseData::TransactionResponse {
+                resource_id: ResponseId::NoResponseId,
+                redirection_data: Some(RedirectForm::Form {
+                    endpoint: item.data.request.get_complete_authorize_url()?,
+                    method: common_utils::request::Method::Get,
+                    form_fields: HashMap::from([
+                        (
+                            "reference".to_string(),
+                            item.response.reference.unwrap_or("".to_string()),
```

Comment:
If reference is a critical fields as if its not available you fail the payment in complete_authorize, its better to fail it here itself

Reply:
Done